pkg/cmd/describe/describe_rollout.go

```diff
@@ -472,8 +514,116 @@ func (o *DescribeRolloutOptions) printTrafficRouting(trafficRouting []rolloutsap
+		fmt.Fprintf(o.Out, tableFormat, " Step:", strconv.Itoa(int(rollout.Status.CanaryStatus.CurrentStepIndex))+"/"+strconv.Itoa(len(canary.Steps)))
+		fmt.Fprint(o.Out, " Steps:\n")
+		currentStepIndex := int(rollout.Status.CanaryStatus.CurrentStepIndex)
+		for i, step := range canary.Steps {
+			isCurrentStep := (i + 1) == currentStepIndex
+			if isCurrentStep {
+				fmt.Fprint(o.Out, "\033[32m")
+			}
+
+			if step.Replicas != nil {
+				fmt.Fprintf(o.Out, tableFormat, "  -  Replicas: ", step.Replicas)
+			}
+			if step.TrafficRoutingStrategy.Weight != nil {
+				fmt.Fprintf(o.Out, tableFormat, "     Weight: ", *step.TrafficRoutingStrategy.Weight)
+			}
+
+			if step.Matches != nil {
+				fmt.Fprintln(o.Out, "     Matches: ")
+				for _, match := range step.Matches {
+					fmt.Fprintln(o.Out, "      - Headers: ")
+					for _, path := range match.Headers {
+						fmt.Fprintf(o.Out, tableFormat, "       - Name:", path.Name)
+						fmt.Fprintf(o.Out, tableFormat, "         Value:", path.Value)
+						fmt.Fprintf(o.Out, tableFormat, "         Type:", *path.Type)
+					}
+				}
+			}
+
+			if isCurrentStep {
+				fmt.Fprintf(o.Out, tableFormat, "     State:", rollout.Status.CanaryStatus.CurrentStepState)
+				fmt.Fprint(o.Out, "\033[0m") // Reset color
+			}
+		}
+	}
+
+	rolloutWorkload := RolloutWorkloadRef{
+		Kind:            rollout.Spec.ObjectRef.WorkloadRef.Kind,
+		Name:            rollout.Spec.ObjectRef.WorkloadRef.Name,
+		StableRevision:  rollout.Status.CanaryStatus.StableRevision,
+		CanaryRevision:  rollout.Status.CanaryStatus.CanaryRevision,
+		PodTemplateHash: rollout.Status.CanaryStatus.PodTemplateHash,
+	}
+
+	// Workload
+	info, err := o.GetResources(rolloutWorkload)
+	if err != nil {
+		fmt.Fprintf(o.Out, "Error getting resources: %v\n", err)
+		return
+	}
+
+	// Images
+	for i, image := range info.Images {
+		if i == 0 {
+			fmt.Fprintf(o.Out, tableFormat, "Images:", image)
+		} else {
+			fmt.Fprintf(o.Out, tableFormat, "", image)
+		}
+	}
+
+	// Revisions
+	fmt.Fprintf(o.Out, tableFormat, "Current Revision:", info.CurrentRevision)
+	fmt.Fprintf(o.Out, tableFormat, "Update Revision:", info.UpdateRevision)
+
+	// Replicas
+	if rollout.Status.ObservedGeneration == rollout.GetObjectMeta().GetGeneration() {
+		fmt.Fprint(o.Out, "Replicas:\n")
+		fmt.Fprintf(o.Out, tableFormat, " Desired:", info.Replicas.Desired)
+		fmt.Fprintf(o.Out, tableFormat, " Updated:", info.Replicas.Updated)
+		fmt.Fprintf(o.Out, tableFormat, " Current:", info.Replicas.Current)
+		fmt.Fprintf(o.Out, tableFormat, " Ready:", info.Replicas.Ready)
+		fmt.Fprintf(o.Out, tableFormat, " Available:", info.Replicas.Available)
+	}
+
+	if len(info.Pod) > 0 {
+		w := tabwriter.NewWriter(o.Out, 0, 0, 2, ' ', 0)
+		fmt.Fprintln(w, "NAME\tREADY\tBATCH ID\tREVISION\tAGE\tRESTARTS\tSTATUS")
+
+		for _, pod := range info.Pod {
+			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s %s\n",
+				pod.Name,
+				pod.Ready,
+				pod.BatchID,
+				pod.Revision,
+				pod.Age,
```

Comment:
when calculate age, currently we will use the first container to calculate the lastTerminationTime, which is not correctly in multi-container cases. we should use the most recent lastTerminationTime among all containers( largest among all pod.Status.ContainerStatuses[*].LastTerminationState.Terminated.FinishedAt.Time)

Reply:
To calculate the age, we follow the same pattern used by kubectl.
`Age:  duration.HumanDuration(time.Since(pod.CreationTimestamp.Time)),`